src/tests/performance/get_blocks.rs

```diff
@@ -0,0 +1,192 @@
+    // │     400 │        100 │          0 │         19 │              2 │          1 │          1 │          2 │          2 │          5 │          99.59 │       4.41 │      9028.24 │
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │     500 │        100 │          0 │         14 │              2 │          1 │          1 │          2 │          2 │          5 │          99.17 │       7.39 │      6706.93 │
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │     600 │        100 │          0 │         14 │              2 │          1 │          1 │          2 │          2 │          6 │          99.00 │       7.46 │      7961.44 │
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │     700 │        100 │          0 │         11 │              2 │          1 │          1 │          2 │          3 │          6 │          98.77 │      10.43 │      6629.62 │
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │     800 │        100 │          0 │         11 │              1 │          1 │          1 │          2 │          2 │          5 │          99.02 │       7.62 │     10396.44 │
+    // └─────────┴────────────┴────────────┴────────────┴────────────────┴────────────┴────────────┴────────────┴────────────┴────────────┴────────────────┴────────────┴──────────────┘
+    // *NOTE* run with `cargo test --release  tests::performance::get_blocks -- --nocapture`
+    // Before running test generate dummy devices with different ips using toos/ips.py
+
+    let synth_counts = vec![1, 50, 100, 200, 300, 400, 500, 600, 700, 800];
+
+    let mut table = RequestsTable::default();
+
+    for synth_count in synth_counts {
+        let target = TempDir::new().expect("couldn't create a temporary directory");
+        let mut node = Node::builder()
+            .build(target.path())
+            .expect("unable to build the node");
+        node.start().await;
+
+        let node_addr = node.net_addr().expect("network address not found");
+
+        let mut synth_sockets = Vec::with_capacity(synth_count);
+        let mut ips = IPS.to_vec();
+
+        for _ in 0..synth_count {
+            // If there is address for our thread in the pool we can use it.
+            // Otherwise we'll not set bound_addr and use local IP addr (127.0.0.1).
+            let ip = if let Some(ip_addr) = ips.pop() {
+                ip_addr
+            } else {
+                "127.0.0.1"
+            };

```

Comment:
aha, nice! now we can do [this](https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or):
```suggestion
            let ip = ips.pop().unwrap_or("127.0.0.1");
```

(or merge this unwrap_or approach with line 94 (whatever you like more)

Reply:
Changed.